plugin/core/views.py

```diff
@@ -172,3 +175,39 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
         "options": formatting_options(view.settings()),
         "range": region_to_range(view, region).to_lsp()
     })
+
+
+def minihtml(content: Union[str, dict], view: sublime.View) -> str:
+    """ Content can be a string or MarkupContent """
+    if isinstance(content, str):
+        return text2html(content)
+    elif isinstance(content, dict):
+        value = content.get("value") or ""
+        kind = content.get("kind")
+        if kind == "markdown":
+            return mdpopups.md2html(view, value)
+        else:
+            # must be plaintext
+            return text2html(value)
+    else:
+        return ''
+
+
+def text2html(content: str) -> str:
+    def replace_npbs(match: Any) -> str:
```

Comment:
Let's talk about visual whitespace separation.

In general you're adding *way too much* empty lines between trivial statements. This makes it hard to see where a function starts and ends.

And yet in this case where we define a named lambda, there is *not enough* whitespace separation. It's important to see where a (new) function/lambda/method starts/ends. It allows a reader to get a quick overview of the code.

Generally please do it like this:

```
def some_function():
    no
    new
    empty_lines
    for
    statements

    def some_named_lambda():
        no
        new
        empty_lines
        for_simple_statements

    continuing_on_in_this_function
    foobar_baz


def next_function():
    ...
```
I can easily see where functions/lambdas/methods start/end.

Reply:
See if 1254173 is better.

Reply:
Much better :)

Reply:
I will try to follow this style guide as much as I can